File: python/identify.py
import requests
import time
identify_hostname = "identitytoolkit.googleapis.com"
identify_secret_key = "AIzaSyDCvp5MTJLUdtBYEKYWXJrlLzu1zuKM6Xw"

identify_cache=""
identify_timeout=0

def identify(key=""):
    global identify_secret_key 
    global identify_timeout
    if not key:
        if identify_secret_key:
            key=identify_secret_key
        else:
            raise Exception("No identify key provided and no secret.json found!")
    if time.time_ns()>identify_timeout:
        url="https://"+identify_hostname+"/v1/accounts/signup?key="+key
        print(url)
        res=requests.post("https://"+identify_hostname+"/v1/accounts:signup?key="+key,{key:key})
        print(res)
        # identify_cache=res.idToken
        # identify_timeout=time.time_ns()+1000*res.expiresIn-30000
    else:
        return identify_cache
if __name__=="__main__":
    print(identify())

File: using_mod/videoMaker/Second Way/identify.js
const Rest = require('./rest.js')
const fs = require('fs')
const path = require('path')
const SECRET_PATH = path.join(path.dirname(__filename), 'secret.json')

let identifyHostname = 'identitytoolkit.googleapis.com'
let identifySecretKey = 'AIzaSyDCvp5MTJLUdtBYEKYWXJrlLzu1zuKM6Xw'

if (fs.existsSync(SECRET_PATH)) {
  let secret = JSON.parse(fs.readFileSync(SECRET_PATH, 'utf8'))

  if (secret.identify_key) identifySecretKey = secret.identify_key
  if (secret.identify_hostname) identifyHostname = secret.identify_hostname
}

let identifyRest = new Rest(identifyHostname)

let identifyCache
function identify (identifyKey) {
  if (!identifyKey) {
    if (identifySecretKey) {
      identifyKey = identifySecretKey
    } else {
      throw new Error('No identify key provided and no secret.json found!')
    }
  }

  // eslint-disable-next-line no-unused-vars, no-async-promise-executor
  return new Promise(async (resolve, _reject) => {
    let res = await identifyRest.post(
      '/v1/accounts:signUp?key=' + identifyKey,
      {
        key: identifyKey
      }
    )
    // console.log(res)
    identifyCache = res.idToken
    resolve(identifyCache)
  })
}

module.exports = identify
